dagshub/data_engine/model/query_result.py

```diff
@@ -548,6 +549,7 @@ def predict_with_mlflow_model(
         post_hook: Callable[[Any], Any] = identity_func,
         batch_size: int = 1,
         log_to_field: Optional[str] = None,
+        is_prediction: Optional[bool] = True
     ) -> Dict[str, Any]:
```

Comment:
Having a `is_prediction` kwarg on a `predict_with_...` function is IMO really confusing API design.
I won't put my foot down and I will push this through for whatever you need right now, but we will probably need to change this in the near future if you want people to actually be using this.

Either way, thanks for adding the docstring

Reply:
Extracted the main code into a `ds._infer_with_mlflow_on_files`, gotten rid of the kwarg in the original functions and instead just call this extracted function